core/templates/dev/head/components/top_navigation_bar/TopNavigationBarDirective.js

```diff
@@ -95,13 +95,92 @@ oppia.directive('topNavigationBar', [function() {
         }
 
         $scope.windowIsNarrow = windowDimensionsService.isWindowNarrow();
+        var handleOverflowResizeTimer;
+        var initialWindowWidth = windowDimensionsService.getWidth();
+        var hideNavElements = {
+          I18N_TOPNAV_DONATE: false,
+          I18N_TOPNAV_ABOUT: false,
+          I18N_CREATE_EXPLORATION_CREATE: false,
+          I18N_TOPNAV_LIBRARY: false
+        };
+
         windowDimensionsService.registerOnResizeHook(function() {
           $scope.windowIsNarrow = windowDimensionsService.isWindowNarrow();
           $scope.$apply();
           // Close the sidebar, if necessary.
           SidebarStatusService.closeSidebar();
+          // Limit calls to handleOverflow() to one per 500ms to prevent flicker
+          $timeout.cancel(handleOverflowResizeTimer);
+
+          // If the window is resized larger, try displaying the hidden elements
+          if (initialWindowWidth < windowDimensionsService.getWidth()) {
+            $.each(hideNavElements, function(element, state) {
+              if (state === true) {
+                $('[translate=' + element + ']')
+                  .closest('div, li').css('display', 'block');
+                hideNavElements[element] = false;
+              }
+            });
+          }
+          initialWindowWidth = windowDimensionsService.getWidth();
+          handleOverflowResizeTimer = $timeout(handleOverflow, 500);
         });
 
+        var handleOverflow = function() {
```

Comment:
Renamed and added docstring. I wasn't sure on the formatting of the docstrings for the javascript files though, so I used JSDoc format. I didn't a docstring format listed in the contributing wiki/frontend doc.